### ISSUE TITLE
dom.d: ids indexing

### ISSUE THREAD

**Issue by trikko** [adamdruppe/arsd#109]
Check: https://github.com/adamdruppe/arsd/blob/master/dom.d#L2069

Adding a `DomElement[string] index;` inside document, and keep it updated on document's parsing / editing, could be improve a lot a search by id, using css selector too. 
Something like "#myid div p" (common case, selector starting with #id) could be trivially improved starting the deep search from index["myid"].

Adding custom indexing for other (non-unique?) properties could be useful too.

**Comment by adamdruppe** [comment]
Yeah, this would be a good idea, but it is easier said than done because there's a LOT of methods that can mutate the dom. Ideally, I should probably add some kind of mutation observer events and use that to keep the index up to date... but the code isn't ready for that right now.

What you can do now though is to maintain your own map. Like on the dpldocs search, I don't modify the dom, just search. so on load, I build such a map to help lookups after.
